pastastore/base.py

```diff
@@ -14,19 +19,46 @@
 from tqdm.auto import tqdm
 
 from pastastore.util import ItemInLibraryException, _custom_warning, validate_names
-from pastastore.version import PASTAS_LEQ_022
+from pastastore.version import PASTAS_GEQ_150, PASTAS_LEQ_022
 
 FrameorSeriesUnion = Union[pd.DataFrame, pd.Series]
 warnings.showwarning = _custom_warning
 
 
+# def weak_lru(maxsize=128, typed=False):
+#     """LRU Cache decorator that keeps a weak reference to 'self'.
+
+#     From https://stackoverflow.com/a/68052994/10596229.
+
+#     Parameters
+#     ----------
+#     maxsize : int, optional
+#         maximum size of cache, by default 128
+#     typed : bool, optional
+#         whether to differentiate between types, by default False
+
+#     """
+
+#     def wrapper(func):
+#         @functools.lru_cache(maxsize, typed)
+#         def _func(_self, *args, **kwargs):
+#             return func(_self(), *args, **kwargs)
+
+#         @functools.wraps(func)
+#         def inner(self, *args, **kwargs):
+#             return _func(weakref.ref(self), *args, **kwargs)
+
+#         return inner
+
+#     return wrapper
+
+
```

Comment:
We wan't to keep this?

Reply:
I guess it would be neater to put this code in a separate branch or issue... I'll do that.